internal/adapters/cloud/aws/autoscaling/adapt.go

```diff
@@ -0,0 +1,285 @@
+package autoscaling
+
+import (
+	"fmt"
+
+	"github.com/aquasecurity/defsec/internal/adapters/cloud/aws"
+	"github.com/aquasecurity/defsec/pkg/concurrency"
+	"github.com/aquasecurity/defsec/pkg/providers/aws/autoscaling"
+	"github.com/aquasecurity/defsec/pkg/state"
+	"github.com/aquasecurity/defsec/pkg/types"
+	"github.com/aws/aws-sdk-go-v2/aws/arn"
+	api "github.com/aws/aws-sdk-go-v2/service/autoscaling"
+	aatypes "github.com/aws/aws-sdk-go-v2/service/autoscaling/types"
+)
+
+type adapter struct {
+	*aws.RootAdapter
+	api *api.Client
+}
+
+func init() {
+	aws.RegisterServiceAdapter(&adapter{})
+}
+
+func (a *adapter) Provider() string {
+	return "aws"
+}
+
+func (a *adapter) Name() string {
+	return "autoscaling"
+}
+
+func (a *adapter) Adapt(root *aws.RootAdapter, state *state.State) error {
+	a.RootAdapter = root
+	a.api = api.NewFromConfig(root.SessionConfig())
+
+	var err error
+	state.AWS.Autoscaling.AutoscalingGroupsList, err = a.getAutoscaling()
+	if err != nil {
+		return err
+	}
+
+	state.AWS.Autoscaling.LaunchConfigurations, err = a.getLaunchConfigurations()
+	if err != nil {
+		return err
+	}
+
+	state.AWS.Autoscaling.NotificationConfigurations, err = a.getNotificationConfigurations()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (a *adapter) getAutoscaling() ([]autoscaling.AutoscalingGroupsList, error) {
+	a.Tracker().SetServiceLabel(" Availability Zones List...")
+
+	var input api.DescribeAutoScalingGroupsInput
+	var autoscalingapi []aatypes.AutoScalingGroup
+
+	for {
+		output, err := a.api.DescribeAutoScalingGroups(a.Context(), &input)
+		if err != nil {
+			return nil, err
+		}
+		autoscalingapi = append(autoscalingapi, output.AutoScalingGroups...)
+
+		a.Tracker().SetTotalResources(len(autoscalingapi))
+		if output.NextToken == nil {
+			break
+		}
+		input.NextToken = output.NextToken
+
+	}
+	a.Tracker().SetServiceLabel("Adapting analyzers...")
+	return concurrency.Adapt(autoscalingapi, a.RootAdapter, a.adaptautoscaling), nil
+
+}
+
+func (a *adapter) adaptautoscaling(autoscalingapi aatypes.AutoScalingGroup) (*autoscaling.AutoscalingGroupsList, error) {
+
+	if autoscalingapi.AutoScalingGroupARN == nil {
+		return nil, fmt.Errorf("missing arn")
+	}
+	parsed, err := arn.Parse(*autoscalingapi.AutoScalingGroupARN)
+	if err != nil {
+		return nil, fmt.Errorf("invalid arn: %w", err)
+	}
+	if parsed.Region != a.Region() {
+		return nil, nil // skip other regions
+	}
+
+	metadata := a.CreateMetadataFromARN(*autoscalingapi.AutoScalingGroupARN)
+	var name string
+	if autoscalingapi.AutoScalingGroupName != nil {
+		name = *autoscalingapi.AutoScalingGroupName
+	}
+
+	var AVZone []types.StringValue
+	for _, AV := range autoscalingapi.AvailabilityZones {
```

Comment:
variable names are usually lowercase in Go.
```suggestion
	for _, av := range autoscalingapi.AvailabilityZones {
```